src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -91,20 +105,31 @@ public void renewClusterSlots(Connection jedis) {
           }
         }
 
+        // Then, we use startNodes to try, as long as startNodes is available,
+        // whether it is vip, domain, or physical ip, it will succeed.
+        if (startNodes != null) {
+          ArrayList<HostAndPort> startNodeList = new ArrayList<>(startNodes);
+          Collections.shuffle(startNodeList);
+          for (HostAndPort hostAndPort : startNodeList) {
```

Comment:
What if we put _attempted_ `hostAndPort`s in a Set and skip those when attempting from getShuffledNodesPool()?

Reply:
Agreed, so that during the `getShuffledNodesPool` stage, nodes that have already been tried and failed can be ignored.
I think, instead of using a new variable, we can use `startNodes` directly. Because once it reaches `getShuffledNodesPool`, it means that `startNodes` has been tried all, and none of them have succeeded.

Reply:
already updated code.